hconfig_utils/mapl3g_hconfig_getter.F90

```diff
@@ -0,0 +1,235 @@
+#include "MAPL_ErrLog.h"
+module mapl3g_hconfig_getter
+
+   use :: pflogger, only: logger_t => logger
+   use :: esmf, MAXSTRLEN => ESMF_MAXSTR
+   use mapl_ErrorHandling
+
+   implicit none
+   public :: HConfigGetter
+   public :: get_value
+
+   type :: HConfigGetter
+      type(ESMF_HConfig) :: hconfig
+      character(len=:), allocatable :: label
+      logical :: found = .FALSE.
+      character(len=:), allocatable :: typestring
+      character(len=:), allocatable :: valuestring
+      logical :: value_equals_default = .FALSE.
+   end type HConfigGetter
+
+   interface get_value
+      module procedure :: get_value_i4
+   end interface get_value
+
+   character(len=*), parameter :: DEFAULT_FORMAT_STRING = '(G0)'
+   character(len=*), parameter :: EMPTY_STRING = ''
+
+   interface HConfigGetter
+      module procedure :: construct
+   end interface HConfigGetter
+
+contains
+   
+   type(HConfigGetter) function construct(hconfig, label, found)
+      type(ESMF_HConfig), intent(in) :: hconfig
+      character(len=*), intent(in) :: label
+      logical, intent(in) :: found
+
+      construct%hconfig=hconfig
+      construct%label=label
+      construct%found=found
+      construct%typestring=EMPTY_STRING
+      construct%valuestring=EMPTY_STRING
+
+   end function construct
+
+   subroutine get_value_i4(getter, value, default, rc)
+      integer(kind=ESMF_KIND_I4), intent(out) :: value !macro VALTYPE
+      character(len=*), parameter :: fmt_ = DEFAULT_FORMAT_STRING !macro FMTSTR
+      integer(kind=ESMF_KIND_I4) :: default_ !macro VALTYPE
+      type(HConfigGetter), intent(inout) :: getter
+      class(*), optional, intent(in) :: default
+      integer, optional,intent(out) :: rc
+      integer :: status = 0
+      character(len=MAXSTRLEN) :: buffer
+
+      getter%typestring = 'TYPESTRING_'
```

Comment:
I think there is a problem here.   Macros don't expand inside of quotes - at least on some compilers.   But I could be wrong.  Usually this is done by making the macro have the quotes:
```
#define TYPESTRING_ 'I4'
```

Reply:
But this is not even parameterized code, so ... not sure what is supposed to be here.

Reply:
Fixed